src/org/traccar/protocol/AdmProtocolDecoder.java

```diff
@@ -119,4 +115,41 @@ protected Object decode(
         return null;
     }
 
+    private Position parseCommandResponse(Channel channel, SocketAddress remoteAddress, ChannelBuffer buf) {
+        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress);
+        if (deviceSession == null) {
+            return null;
+        }
+
+        Position position = new Position();
+        position.setProtocol(getProtocolName());
+        position.setDeviceId(deviceSession.getDeviceId());
+
+        getLastLocation(position, null);
+
+        position.set(Position.KEY_RESULT, buf.readBytes(129).toString(StandardCharsets.UTF_8));
```

Comment:
We probably need to trim this string. Also, can we re-use `CMD_RESPONSE_SIZE` constant here?

Reply:
129 bytes is fixed length for response text. We can use (CMD_RESPONSE_SIZE - 3) if its important.

Reply:
Yes, that's what I meant. Can you please also add a unit test for it.

Reply:
I checked some commands and i think that simple trim is not enough. Device dont truncate previous command result,
Example:
Exec STATUS: 
"result":"ID=1 Soft=0x51 GPS=919 Time=22:36:21 09.09.17 Val=0 Lat=61.6778 Lon=50.8242 V=0 SatCnt=3+8 Stat=0x0010 In_alarm=0x00\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000"
Then exec INPUT 0: 
"result":"INPUT0 = 0\u0000x51 GPS=919 Time=22:36:21 09.09.17 Val=0 Lat=61.6778 Lon=50.8242 V=0 SatCnt=3+8 Stat=0x0010 In_alarm=0x00\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000"
Input command result rewrote beginning of previous command with trailing \u0000 and not truncate remaining buffer. 
Maybe better not trim but get string before first \u0000 symbol?

Reply:
Yes. Looks like it's a zero-terminated string, so we need to properly cut it.